### DISCUSSION TITLE
Can I set custom value when update an entity?

### DISCUSSION THREAD

**Discussion by lesomnus** [ent/ent#2776] [Q&A]
Let's say I have an entity with two fields `CreateAt` and `ProcessedAt` with go type `time.Time`.
I want to use DB (postgres) time for consistency, so I use annotation, `Default: "now()"` for `CreateAt`.
After do something, now I want to update the field `ProcessedAt` with DB time using `now()` but there is no such method that accept raw sql string on generated code.
How can I solve this problem?

**Comment by a8m** [comment]
Hey @lesomnus 👋 

You can use the `entsql.Annotation` to inline database-specific `DEFAULT` expressions. For example:

```go
field.Time("created_at").
	Default(time.Now).
	Annotations(&entsql.Annotation{
		Default: "now()",
	})
```

**Reply by lesomnus** [reply]
Thanks for the quick reply!
Can I use `now()` when updating the `ProcessedAt` field ?

saying,
```go
client.Entity.UpdateOneID(id).
		SetProcessedAt("???"). // how to set raw sql "now()"??
		Save(ctx)
```

**Reply by a8m** [reply]
> Can I use now() when updating the ProcessedAt field?

The `now()` is set as the `DEFAULT` value. i.e. it won't be updated by the database in case you updated it from Go.

**Reply by lesomnus** [reply]
Ahhaa! Thanks!

**Comment by qazwsxedckll** [comment]
@a8m Hi, I think what he meant is to add ON UPDATE in the sql at migration.
Something like 'ProcessedAt' timestamp NULL DEFAULT CURRENT_TIMESTAMP **ON UPDATE CURRENT_TIMESTAMP**
I run into the same problem because someone else is update the table with raw sql.

**Reply by lesomnus** [reply]
At that time I thoght that if I set null in the update query, the default value is used. But it didn't, so it didn't answer my question.
The intent of my question was how to write a query like this:
```sql
UPDATE some_table set processed_at = now() WHERE id = 42
```
where `now()` is postgres function that returns current system time of DB.

**Comment by lesomnus** [comment]
I think this is now possible by `entsql.Default` with `Annotation` as described in https://entgo.io/docs/schema-fields#default-values
Thanks!

**Comment by dev-tathkarah** [comment]
```go
func (CommonMixin) Fields() []ent.Field {
	return []ent.Field{
		field.Time("created_at").Immutable().Default(time.Now).StructTag(`json:"createdAt"`),
		field.Time("modified_at").Default(time.Now).UpdateDefault(time.Now).StructTag(`json:"modifiedAt"`),
	}
}
```

As i understand from message from @a8m . I could change created_at in a way that

```go
field.Time("created_at").Immutable().
Default(time.Now).Annotations(&entsql.Annotation{Default: "now()",}).StructTag(`json:"createdAt"`),
```

but how to change the field modified_at?

 
Any assistance is higly appreciated
